wordpress-mobile/WordPress-iOS
# Recommend App: Add Tracks

Refs #16995, `pctCYC-8X-p2#comment-340`
Depends on #17020

## Summary

This adds tracking events in the `ShareAppContentPresenter`, with details as described in `pctCYC-8X-p2#comment-340`. Additionally, I updated `TestAnalyticsTracker` to store `trackString` calls (for unit testing purposes).

**Note:** I'll keep this PR in open until the middle of next week, in case if there are any feedback on the event names, etc.

## To Test
There's no visual change, but tests for tracking verification is added. Make sure the tests are ✅ !

## Regression Notes
1. Potential unintended areas of impact
None.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Added unit tests for the tracking implementation.

3. What automated tests I added (or what prevented me from doing so)
Added unit tests for the tracking implementation.

PR submission checklist:
